React

brettz9 opened the pull request:
test: coverage for React

I attempted here to build out what the structure of the expected arguments should be (to the limited extent I could figure them out), but I'd appreciate any improvements on what usage should be.

Also:
- change from deprecated husky command

Btw, your GitHub project description still says, "The missing zod (de)?serialization library".

Gregoor commented:
Thanks (again^23)! 

As a side-comment, at some point I'd love to factor React out into a separate packages and publish zodex without it. In my mind that's the last blocker for v1.0

> Btw, your GitHub project description still says, "The missing zod (de)?serialization library".

That was meant to be read as Regex, e.g. the `(de)` group is made optional through `?`. But I see how it's confusing, so I changed it 🙃

brettz9 commented:
> Thanks (again^23)!
> 
> As a side-comment, at some point I'd love to factor React out into a separate packages and publish zodex without it. In my mind that's the last blocker for v1.0

Sure. For now, did you want this one? I've rebased.

> > Btw, your GitHub project description still says, "The missing zod (de)?serialization library".
> 
> That was meant to be read as Regex, e.g. the `(de)` group is made optional through `?`. But I see how it's confusing, so I changed it 🙃

Ahh. Could just be me though! :-)

Gregoor commented:
> Sure. For now, did you want this one? I've rebased.


Thanks, that's indeed what I wanted.

Looking at the PR now I'm wondering though if we could make the test instead explicitly test React rendering.

__

...and then zooming out a bit more, I'm wondering if we should just kick the react dependency out now and keep the `ui.ts` file as an examples around instead. @heikir is commonbase still depending on `zodex`?

Gregoor commented:
I've dropped React in #42

brettz9 commented:
Cool... There's also the `infer.ts` file which is now unused, but I've got that removed in my browser PR if you like...

Gregoor commented:
Oh good point, thanks! I'm still thinking I want to export that so that people can implement the dropped UI code on their side easily
